### PR TITLE
Comments API: hide deleted and hidden comments (plus docs)

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

This PR contains the following two fixes:

- deleted comments will appear like this in the tree of comments

```json
{
    "type_of": "comment",
    "id_code": "16",
    "body_html": "<p>[deleted]</p>",
    "user": {},
    "children": [{}, {}]
 }
```

- hidden comments will appear like this in the tree of comments

```json
{
    "type_of": "comment",
    "id_code": "16",
    "body_html": "<p>[hidden by post author]</p>",
    "user": {},
    "children": [{}, {}]
 }
```

I used the same text we use for the website. Their children comments won't be hidden or deleted, the logic checks each single comment to see if it shouldn't appear in full, like the website does.

I also added the comments API to the documentation.

## Related Tickets & Documents

#4474 at least regarding documentation

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [x] docs.dev.to
- [ ] readme
- [ ] no documentation needed

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,8 +1,12 @@
 class Comment < ApplicationRecord
+  TITLE_DELETED = "[deleted]".freeze
+  TITLE_HIDDEN = "[hidden by post author]".freeze
```
Comment (rhymes) [inline app/models/comment.rb]: refactored this so if we ever change it in the frontend website, it changes in the API as well

File: app/views/api/v0/comments/_comment.json.jbuilder [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+json.type_of "comment"
+json.id_code comment.id_code_generated
 
-json.id_code    comment.id_code_generated
-json.body_html  comment.processed_html
-
-json.partial! "api/v0/shared/user", user: comment.user
+if comment.deleted?
+  json.body_html "<p>#{Comment::TITLE_DELETED}</p>"
+  json.set! :user, {}
```
Comment (rhymes) [inline app/views/api/v0/comments/_comment.json.jbuilder]: fake empty user :)

File: app/models/comment.rb (hunk 1 above)
Comment (citizen428) [reply app/models/comment.rb]: This is why always advocate for using I18n, even in apps that only use one language. Much nicer also for specs, where you can assert on presence of `I18n.t(...)`, so copy can change without unnecessary breakage.

Comment (rhymes) [reply app/models/comment.rb]: Agreed, at some point we should start moving strings outside the code as much as possible :)

We do have a few i18n/l10n issues: https://github.com/thepracticaldev/dev.to/labels/area%3A%20internationalization%2Flocalization

**Comment by rhymes** [comment]
Great catch @fdoxyz!! Writing OpenAPI manually is not super handy and errors like this are bound to happen. I think I'm going to add a validator/linter as a pre-commit hook like we have for other file types

**Pull request merged**